### ISSUE TITLE
time zones not working correctly

### ISSUE THREAD

**Issue by fzipp** [robpike/ivy#158]
I start Ivy and enter:

```
)timezone 
	CEST 0
```

CEST is correct for my location, but offset 0 isn't. CEST is UTC+2, so the offset should be 7200.

Now I switch the time zone:

```
)timezone "CET"
	CET 7200
```

CET is expected, but offset 7200 isn't. CET is UTC+1, so the offset should be 3600.

Now I want to switch back to the original time zone:

```
)timezone "CEST"
	no such time zone: no such time zone
```

Even though Ivy told me this zone name at the beginning, so it should be known.

**Comment by robpike** [comment]
There's a lot going on here, and some of it is caused by the peculiarities of the timezone database. I presume you're using Unix in some form, because otherwise I believe none of this will work.

One thing to understand (it's really subtle, and I am still missing some nuance; read on) is how time.Location works in Go. They are not time zones per se, they are locations in which time can be determined. Have a look:

```
% ivy
)timezone
AEST 36000

)timezone "EST"

)timezone
EST -14400

)timezone "EDT"

)timezone
EDT -14400
```

First, note that my timezone, AEST, prints correctly. That's a property of the system, somehow, and for the life of me I can't figure out where it comes from.

Second, note that it knows about EST and EDT but both print with the same offset, which is actually the _current_ offset to UTC at the location in which EST is a valid time zone, e.g. New York. You can see this happening a bit in the test suite, which asks the time at the Unix epoch. The test has the wrong offset for that date, and will break when the US goes back to standard time. I should probably have commented that, and will return to it then.

Third, I can't explain why you see zero for CEST, but on my machine CEST is not a known zone so I can't get ivy into that state. I can do this though:

```
% ivy
)timezone "GMT"

)timezone
GMT 3600

)timezone "UTC"

)timezone
UTC 0

)timezone "Europe/Paris"

)timezone
Europe/Paris 7200
```

Try that and see what you get. You could also try setting the TZ environment variable and explore what happens then.

Despite this explanation, I do believe there are bugs, I just don't know how to deal with the inconsistencies in the timezone database and its interaction with what the local system says.

**Comment by robpike** [comment]
Thanks for the good report.

I know what's wrong with GMT - I have it mapped to Europe/London instead of UTC - and that's trivial to fix.

The others require more investigation.

**Comment by robpike** [comment]
Yes, you've put your finger on something that's at the heart of this. I plan to investigate with the Go team.

**Comment by robpike** [comment]
https://github.com/golang/go/issues/63345

**Comment by robpike** [comment]
Not obvious what to do here. Closing.
